### PR TITLE
Fix an issue with native engine function extension.

### PR DESCRIPTION
**Pull request by RaananW**

### CODE REVIEW

**Comment by bjsplat** [comment]
Please make sure to label your PR with "bug", "new feature" or "breaking change" label(s).
To prevent this PR from going to the changelog marked it with the "skip changelog" label.

File: packages/dev/core/src/Engines/thinEngine.functions.ts [hunk 1]
```diff
@@ -323,33 +323,6 @@ function _compileRawShader(source: string, type: string, gl: WebGLContext, _cont
...
-        if (uniform) {
-            _boundUniforms[index] = uniform;
-        }
-    }
-    return _boundUniforms;
-}
-
 /**
  * @internal
  */
```
Comment (bghgary) [inline packages/dev/core/src/Engines/thinEngine.functions.ts:328]: I can't add a comment to the right place, so I'll add it here. Shouldn't the `_executeWhenRenderingStateIsCompiled` be moved to `effect.webgl.functions.ts`?

**Comment by bjsplat** [comment]
Snapshot stored with reference name:
refs/pull/15063/merge

Test environment: 
https://babylonsnapshots.z22.web.core.windows.net/refs/pull/15063/merge/index.html

To test a playground add it to the URL, for example:

https://babylonsnapshots.z22.web.core.windows.net/refs/pull/15063/merge/index.html#WGZLGJ#4600

Links to test babylon tools with this snapshot:

https://playground.babylonjs.com/?snapshot=refs/pull/15063/merge
https://sandbox.babylonjs.com/?snapshot=refs/pull/15063/merge
https://gui.babylonjs.com/?snapshot=refs/pull/15063/merge
https://nme.babylonjs.com/?snapshot=refs/pull/15063/merge

To test the snapshot in the playground with a playground ID add it after the snapshot query string:

https://playground.babylonjs.com/?snapshot=refs/pull/15063/merge#BCU1XR#0

File: packages/dev/core/src/Engines/thinEngine.functions.ts (hunk 1 above)
Comment (bghgary) [reply packages/dev/core/src/Engines/thinEngine.functions.ts:328]: Okie

**Comment by bjsplat** [comment]
WebGL2 visualization test reporter:

https://babylonsnapshots.z22.web.core.windows.net/refs/pull/15063/merge/testResults/webgl2playwright/index.html

**Comment by bjsplat** [comment]
Visualization tests for WebGPU (Experimental)
Important - these might fail sporadically. This is an optional test.

https://babylonsnapshots.z22.web.core.windows.net/refs/pull/15063/merge/testResults/webgpuplaywright/index.html

**Review by deltakosh — APPROVED** [review]

**Pull request merged**